test/cql-pytest/test_empty.py

```diff
@@ -218,3 +218,74 @@ def test_empty_string_for_nonstring_partition_key2(cql, test_keyspace):
+# that it is working as expected.
+def test_empty_string_in_collection(cql, test_keyspace):
+    schema = 'p int primary key, v list<text>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        p = unique_key_int()
+        cql.execute(f"INSERT INTO {table} (p,v) VALUES ({p}, ['hi',''])")
+        assert list(cql.execute(f"SELECT * FROM {table} WHERE p={p}")) == [(p, ['hi',''])]
+        # Prepared version of the same test.
+        p = unique_key_int()
+        stmt = cql.prepare(f"INSERT INTO {table} (p,v) VALUES ({p}, ?)")
+        cql.execute(stmt, [['hello','']])
+        assert list(cql.execute(f"SELECT * FROM {table} WHERE p={p}")) == [(p, ['hello',''])]
+
+def test_empty_int_in_collection(cql, test_keyspace):
+    schema = 'p int primary key, v list<int>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        p = unique_key_int()
+        cql.execute(f"INSERT INTO {table} (p,v) VALUES ({p}, [1, blobAsInt(0x)])")
+        assert list(cql.execute(f"SELECT * FROM {table} WHERE p={p}")) == [(p, [1,None])]
+        # There is no official way to do this same test with a prepared
+        # statment because there is no official way to pass an empty
+        # integer value. There is an unofficial way (and arguably buggy) by
+        # passing None - see the next test:
+
+# Null values are not allowed inside collections, and as we see below
+# indeed a non-prepared statement attempted to put a null in a list is
+# rejected. But in a prepared statement, because of a peculiarity or even
+# bug of the Python driver, a None value in a list is converted to an
+# *empty* value (value of length 0), instead of the expected null value
+# (length -1), so a None in a list should behave like an empty value in
+# test_empty_*_in_collection above. Reproduces #5825.
+def test_null_string_in_collection(cql, test_keyspace):
+    schema = 'p int primary key, v list<text>'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        p = unique_key_int()
+        with pytest.raises(InvalidRequest, match='null is not supported inside collections'):
+            cql.execute(f"INSERT INTO {table} (p,v) VALUES ({p}, ['hi', null])")
+        # Prepared-statement version of the same test. Because of a driver
+        # bug, the None below is *not* converted to a null as a expected,
+        # but rather to an empty string. So the expected behavior from Scylla
+        # and Cassandra is not to print "null is not supported" but to put an
+        # actual empty string in the list.
+        # If the driver's behavior here ever changes (fixed?), this test will
+        # need to be changed as well (and probably moved to test_null.py).
+        p = unique_key_int()
+        stmt = cql.prepare(f"INSERT INTO {table} (p,v) VALUES ({p}, ?)")
+        cql.execute(stmt, [['hello',None]])
+        assert list(cql.execute(f"SELECT * FROM {table} WHERE p={p}")) == [(p, ['hello',''])]
```

Comment:
It bothers me a bit that we have tests which depend on buggy driver behavior, but I guess the comment explains everything clearly.

Reply:
Right, I'm not very happy about this as well :-( But I'm not sure anybody is planning to fix (or even report...) this bug any time soon, so it may stay like this for a long time. If somebody wants to fix this soon, we can hold the test until it is done.
By the way, even when the driver is finally fixed, I want pytest to work regardless of which driver version you have, so we'll need to skip the test on old driver versions (like I do in conftest.py in driver_bug_1 - we'll now have a new driver_bug_2).

Reply:
If there's an open bug on the driver, it'd be great to add it to the comment above. If there isn't, perhaps there should be one?

Reply:
To be honest, I'm a bit confused what is our plan for the CQL driver. In such cases a case of bug, should I create an issue in Datastax's driver and we will later merge their fix, or did we already diverged in the driver and I should focus on reporting (and fixing) only Scylla's driver?

Reply:
I hopened https://github.com/scylladb/python-driver/issues/201. I don't know if we want to report issues like that, or should report them in Datastax's JIRA instead.